src/compiler/emitter.ts

```diff
@@ -4215,15 +4219,19 @@ var __param = this.__param || function(index, decorator) { return function (targ
                     return emitOnlyPinnedOrTripleSlashComments(node);
                 }
 
-                emitStart(node);
-                if (isES6ExportedDeclaration(node)) {
-                    write("export ");
+                if (!isModuleMergedWithES6Class(node)) {
+                    emitStart(node);
+                    if (isES6ExportedDeclaration(node)) {
+                        write("export ");
+                    }
+
+                    write("var ");

```

Comment:
Oh interesting. I did not realize that in ES6, you can export a var, and then reassign to it, and it will amend the export binding too.